larreco/QuadVtx/EvalVtx_module.cc

```diff
@@ -0,0 +1,135 @@
+// Chris Backhouse - c.backhouse@ucl.ac.uk - Oct 2019
+
+#include "fhiclcpp/ParameterSet.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Core/EDAnalyzer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art_root_io/TFileService.h"
+#include "canvas/Persistency/Common/FindManyP.h"
+
+#include "lardataobj/RecoBase/Vertex.h"
+#include "lardataobj/RecoBase/PFParticle.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+
+#include "TTree.h"
+
+namespace{
+  int gEvt;
+  double gTrueX, gTrueY, gTrueZ;
+  std::vector<double> gRecoX, gRecoY, gRecoZ;
+}
```

Comment:
If a user wants to configure two instances of the `EvalVtx`, then these global variables will be shared between the two module instances and the `TTree` constructed in one of the modules can easily be malformed.  Please make the variables members of the `EvalVtx` class to avoid this problem.

Reply:
well, only in the case of multi-threading. But fine, done.